### PR TITLE
Make lintmodern_cni required, and retire classic lint_cni.

### PR DESCRIPTION
**Pull request by geeknoid**
I'll eventually rename the job to drop "modern".

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,6 +10,8 @@ require (
 	github.com/google/go-github v17.0.0+incompatible
 	github.com/hashicorp/go-multierror v0.0.0-20171204182908-b7773ae21874
 	github.com/kr/pretty v0.1.0
+	github.com/onsi/ginkgo v1.7.0 // indirect
```
Comment (fejta) [inline go.mod]: This change seems unrelated?

Comment (geeknoid) [reply go.mod]: The tool chain did this change. It's becauee an earlier PR from John removed this since he's testing out go 1.13. Since I'm still on 1.12, that version of the compiler wants those two imports. When we switch everything to 1.13, then we'll refresh all the go.mod files.

Comment (fejta) [reply go.mod]: Then please put that into its own PR. It is unrelated to this change.

Comment (fejta) [reply go.mod]: Note that bazel avoids these sorts of problems, by declaring -- and using -- a particular go version.

**Comment by geeknoid** [comment]
PTAL.

**Review by howardjohn — APPROVED** [review]

**Comment by rlenglet** [comment]
`make lint_modern` fails in the `release-1.3` branch. No PR can be merged into the `release-1.3` branch now, e.g. https://github.com/istio/cni/pull/173#issuecomment-532016382.
Can this be reverted?

**Comment by rlenglet** [comment]
Or is it missing a common files update?

**Comment by rlenglet** [comment]
Updating the common files doesn't fix that: https://github.com/istio/cni/pull/174.
